mmdet/datasets/dataset_wrappers.py

```diff
@@ -1,282 +1,336 @@
+                contains the categories whose frequency below the threshold,
+                it would be repeated.
+
+        Returns:
+            list[float]: The repeat factors for each images in the dataset.
+        """
+
+        # 1. For each category c, compute the fraction # of images
+        #   that contain it: f(c)
+        category_freq = defaultdict(int)
+        num_images = len(dataset)
+        for idx in range(num_images):
+            cat_ids = set(self.dataset.get_cat_ids(idx))
+            if len(cat_ids) == 0 and not self.filter_empty_gt:
+                cat_ids = set([len(self.CLASSES)])
+            for cat_id in cat_ids:
+                category_freq[cat_id] += 1
+        for k, v in category_freq.items():
+            category_freq[k] = v / num_images
+
+        # 2. For each category c, compute the category-level repeat factor:
+        #    r(c) = max(1, sqrt(t/f(c)))
+        category_repeat = {
+            cat_id: max(1.0, math.sqrt(repeat_thr / cat_freq))
+            for cat_id, cat_freq in category_freq.items()
+        }
+
+        # 3. For each image I, compute the image-level repeat factor:
+        #    r(I) = max_{c in I} r(c)
+        repeat_factors = []
+        for idx in range(num_images):
+            cat_ids = set(self.dataset.get_cat_ids(idx))
+            if len(cat_ids) == 0 and not self.filter_empty_gt:
+                cat_ids = set([len(self.CLASSES)])
+            repeat_factor = 1
+            if len(cat_ids) > 0:
+                repeat_factor = max(
+                    {category_repeat[cat_id]
+                     for cat_id in cat_ids})
+            repeat_factors.append(repeat_factor)
+
+        return repeat_factors
+
+    def __getitem__(self, idx):
+        ori_index = self.repeat_indices[idx]
+        return self.dataset[ori_index]
+
+    def __len__(self):
+        """Length after repetition."""
+        return len(self.repeat_indices)
+
+@DATASETS.register_module()
+class MultiImageMixDataset:
+    def __init__(self, dataset, pipelines, dynamic_scale, enable_mixaug=True):
+        assert isinstance(pipelines, collections.abc.Sequence)
+        self.pipelines = []
+        for pipeline in pipelines:
+            if isinstance(pipeline, dict):
+                pipeline = build_from_cfg(pipeline, PIPELINES)
+                self.pipelines.append(pipeline)
+            elif callable(pipeline):
+                self.pipelines.append(pipeline)
+            else:
+                raise TypeError('pipeline must be callable or a dict')
+
+        self.dataset = dataset
+        self.CLASSES = dataset.CLASSES
+        if hasattr(self.dataset, 'flag'):
+            self.flag = dataset.flag
+        self.num_sample = len(dataset)
+
+        self.dynamic_scale = dynamic_scale
+        self.enable_mixaug = enable_mixaug
+
+    def disable_mixaug(self):
+        for pipeline in self.pipelines:
+            if hasattr(pipeline, "get_indexes"):
+                self.pipelines.remove(pipeline)

```

Comment:
If remove the pipeline, It cannot be turn on later. What about skipping the pipeline which has attribute "get_indexes" in __getitem__()?

Reply:
Can’t write like that, it’s not flexible enough